apps/api/src/storage/domain/storage.service.ts

```diff
@@ -0,0 +1,75 @@
+import {Injectable} from '@nestjs/common';
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import {parse} from 'path';
+import {
+    InvalidPathException
+} from './exceptions/InvalidPathExceptions';
+
+@Injectable()
+export class StorageService {
+    private readonly storagePath = 'storage';
+
+    constructor() {
+        if (!fs.existsSync(this.storagePath))
+            fs.mkdirSync(this.storagePath);
+    }
+
+    getFile(folderPath: string): fs.ReadStream {
+        if (!this.isValidPath(folderPath)) throw new InvalidPathException(folderPath);
+        return fs.createReadStream(this.formatPath(folderPath));
+    }
+
+    async createFile(file: Buffer, folderPath: string): Promise<string> {
+        if (!this.isValidPath(folderPath)) {
+            throw new InvalidPathException(folderPath)
+        }
+
+        const filename = randomUUID();
+
+        this.createPathIfMissing(folderPath);
+        await fs.promises.writeFile(this.formatPath(folderPath, filename), file);
+        return filename;
+    }
+
+    async replaceFile(file: Buffer, filePath: string): Promise<string> {
+        if (!this.isValidFolderPath(filePath)) throw new InvalidPathException(filePath);
+
+        await fs.promises.unlink(this.formatPath(filePath));
+        const parsedPath = parse(filePath);
+        return this.createFile(file, parsedPath.dir);
+    }
+
+    formatPath(folderPath: string, filename?: string): string {
+        return `${this.storagePath}${folderPath}${ filename ? `/${filename}` : '' }`;
+    }
+
+    createPathIfMissing(folderPath: string): void {
+        if (!this.isValidFolderPath(folderPath)) throw new InvalidPathException(folderPath);
+        folderPath = this.storagePath + folderPath;
+
+        if (!fs.existsSync(folderPath))
+            fs.mkdirSync(folderPath, {
+                recursive: true
+            });
+    }
+
+    isValidPath(folderPath: string, filename?: string): boolean {
+        if (filename) {
+            return this.isValidFolderPath(folderPath) && this.isValidFileName(filename)
+        } else {
+            const parsedPath = parse(folderPath);
+            return this.isValidFolderPath(parsedPath.dir) && this.isValidFileName(parsedPath.base);
+        }
+    }
```

Comment:
można uniknąć powtórzenia `return this.isValidFolderPath(folderPath) && this.isValidFileName(filename)`